docs/components/generic/_index.md

```diff
@@ -0,0 +1,142 @@
+tags: ["generic", "components"]
+icon: "/icons/components.png"
+images: ["/icons/components.png"]
+no_list: true
+# SMEs:
+---
+
+The _generic_ component type is for custom components that do not fit into any of the other component {{< glossary_tooltip term_id="subtype" text="subtypes" >}}.
+
+There are no built-in generic component models (other than `fake`).
+Generic is only for [modular resource](/extend/modular-resources/) models that don't fit into any of the typical categories.
+
+Use cases:
+
+- You want to implement a new model of component as a modular resource, and your device isn't an arm, a base, a board, a camera, or any of the other [built-in Viam component types](/components/).
+- You are using someone else's modular resource, which they implemented with the generic subtype.
+
+{{% alert title="Important" color="note" %}}
+
+The generic component API only supports the `DoCommand` method.
+If you use the generic subtype, your module needs to define any and all component functionality and pass it through `DoCommand`.
+
+Whenever possible, it is best to use an [existing component API](/components/) instead of generic so that you do not have to replicate code.
+If you want to use most of an existing API but need just a few other functions, try using the `DoCommand` endpoint and extra parameters to add custom functionality to an existing subtype.
+
+{{% /alert %}}
+
+## Configuration
+
+The generic component supports the following models.
+Click on the component model name for detailed configuration information.
+
+<!-- prettier-ignore -->
+Model | Description
+----- | -----------
+[`fake`](fake/) | A model used for testing, with no physical hardware.
+
+If you want to use another generic model with Viam, you can [define a custom component](../../extend/).
+
+## Control your board with Viam's client SDK libraries
+
+To get started using Viam's SDKs to connect to and control your robot, go to your robot's page on [the Viam app](https://app.viam.com), navigate to the **Code sample** tab, select your preferred programming language, and copy the sample code generated.
+
+{{% snippet "show-secret.md" %}}
+
+When executed, this sample code will create a connection to your robot as a client.
+Then control your robot programmatically by getting your `generic` component from the robot with `FromRobot` and adding API method calls, as shown in the following examples.
+
+These examples assume you have a board called "my_board" configured as a component of your robot.
+If your board has a different name, change the `name` in the code.
+
+Be sure to import the generic package for the SDK you are using:
+
+{{< tabs >}}
+{{% tab name="Python" %}}
+
+```python
+from viam.components.generic import Generic
+```
+
+{{% /tab %}}
+{{% tab name="Go" %}}
+
+```go
+import (
+  "go.viam.com/rdk/components/generic"
+)
+```
+
+{{% /tab %}}
+{{< /tabs >}}
+
+## API
+
+The generic component supports the following methods:
```

Comment:
```suggestion
The generic component supports the following method:
```

Reply:
I was wondering whether I should keep it uniform, and in case we ever add more methods (or document more of the general methods) and forget to change this. Will apply.